homeassistant/components/automation/__init__.py

```diff
@@ -782,7 +782,7 @@ def if_action(variables: Mapping[str, Any] | None = None) -> bool:
         for index, check in enumerate(checks):
             try:
                 with trace_path(["condition", str(index)]):
-                    if not check(hass, variables):
+                    if check(hass, variables) is False:
```

Comment:
This flips the existing working of disabled conditions and may lead to unexpected behavior for people currently using these in their existing installations. 

We have no idea what that will trigger and I think therefore we should not do this.

Reply:
I've added a logic table to the original post, hopefully clearing up the current and proposed logic.

Reply:
Thanks for clearing that up. I misunderstood your comment then.

> This flips the existing working of disabled conditions and may lead to unexpected behavior for people currently using these in their existing installations.
> 
> We have no idea what that will trigger and I think therefore we should not do this.

I disagree with this thinking. If two `failing` conditions are within an `OR` group, and one of them is disabled, this should not cause the `OR` group to pass. For example, a user might have an automation as follows:
```
Trigger: Front door opens
Condition Group: OR
  - Condition: Time < 7 AM
  - Condition: Before Sunrise
Action: Turn off house lights
```
As they go throughout the year, their household needs change, and kids are now on summer vacation; they might disable the `Before sunrise` condition temporarily. What was their intention? Their intention/intuition (as proposed code) is that it will continue to activate before 7AM. Their intention wasn't that the automation should always run now, including after sunrise (as current code).

Perhaps even the user is debugging their automation to figure out which condition is causing it to not work like expected.

I understand the hesitance to break people's current automations, but that should not prevent improvements moving forward.

Reply:
>  If two failing conditions are within an OR group, and one of them is disabled, this should not cause the OR group to pass.

That is defending the change and not looking at existing users. I was not discussing the reasoning behind the change, I was pointing out the dangers of this change.

The fact is, this change flips it around, changing existing logic for users. Meaning things that previously didn't pass, now will pass, and the other way around (when using a disabling condition). This change is, therefore, dangerous, as things could all of a sudden start running or stop running unintendedly on upgrade.